sourcerer-esjc/src/main/java/org/elder/sourcerer2/esjc/EventStoreEsjcEventRepository.kt

```diff
@@ -0,0 +1,553 @@
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.github.msemys.esjc.CannotEstablishConnectionException
+import com.github.msemys.esjc.CatchUpSubscription
+import com.github.msemys.esjc.CatchUpSubscriptionListener
+import com.github.msemys.esjc.CatchUpSubscriptionSettings
+import com.github.msemys.esjc.ConnectionClosedException
+import com.github.msemys.esjc.EventStore
+import com.github.msemys.esjc.EventStoreException
+import com.github.msemys.esjc.ResolvedEvent
+import com.github.msemys.esjc.SliceReadStatus
+import com.github.msemys.esjc.StreamEventsSlice
+import com.github.msemys.esjc.SubscriptionDropReason
+import com.github.msemys.esjc.node.cluster.ClusterException
+import com.github.msemys.esjc.operation.AccessDeniedException
+import com.github.msemys.esjc.operation.CommandNotExpectedException
+import com.github.msemys.esjc.operation.InvalidTransactionException
+import com.github.msemys.esjc.operation.NoResultException
+import com.github.msemys.esjc.operation.NotAuthenticatedException
+import com.github.msemys.esjc.operation.ServerErrorException
+import com.github.msemys.esjc.operation.StreamDeletedException
+import com.github.msemys.esjc.operation.WrongExpectedVersionException
+import com.github.msemys.esjc.operation.manager.OperationTimedOutException
+import com.github.msemys.esjc.operation.manager.RetriesLimitReachedException
+import com.github.msemys.esjc.subscription.MaximumSubscribersReachedException
+import com.github.msemys.esjc.subscription.PersistentSubscriptionDeletedException
+import com.google.common.base.Preconditions
+import com.google.common.collect.ImmutableList
+import com.google.common.collect.ImmutableMap
+import org.elder.sourcerer2.EventData
+import org.elder.sourcerer2.EventId
+import org.elder.sourcerer2.EventNormalizer
+import org.elder.sourcerer2.EventRecord
+import org.elder.sourcerer2.EventRepository
+import org.elder.sourcerer2.EventSubscriptionUpdate
+import org.elder.sourcerer2.ExpectedVersion
+import org.elder.sourcerer2.RepositoryReadResult
+import org.elder.sourcerer2.RepositoryVersion
+import org.elder.sourcerer2.StreamId
+import org.elder.sourcerer2.StreamReadResult
+import org.elder.sourcerer2.StreamVersion
+import org.elder.sourcerer2.exceptions.PermanentEventReadException
+import org.elder.sourcerer2.exceptions.PermanentEventWriteException
+import org.elder.sourcerer2.exceptions.RetriableEventReadException
+import org.elder.sourcerer2.exceptions.RetriableEventWriteException
+import org.elder.sourcerer2.exceptions.UnexpectedVersionException
+import org.elder.sourcerer2.utils.ElderPreconditions
+import org.reactivestreams.Publisher
+import org.slf4j.LoggerFactory
+import reactor.core.publisher.Flux
+import reactor.core.publisher.FluxEmitter
+import java.io.IOException
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.ExecutionException
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.TimeoutException
+
+/**
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') One class should have one responsibility. Large classes tend to handle many things at once. Split up large classes into smaller classes that are easier to understand. SIZE 170 is greater than the threshold 150. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.elder.org/coding_rules#rule_key=detekt-kotlin%3ALargeClass)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Classes with many functions tend to do too many things and often come in conjunction with large classes and can quickly become God classes. Consider extracting methods to (new) classes better matching their responsibility. SIZE 27 is greater than the threshold 10. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.elder.org/coding_rules#rule_key=detekt-kotlin%3ATooManyFunctions)